### PR TITLE
refactor: align code to use Editor for source manipulation

### PR DESCRIPTION
**Pull request by MarcinVaadin**
## Description

Alignment Theme Editor java source manipulation on top of Editor prototype introduced in  https://github.com/vaadin/flow/compare/main...feature/edit 

Added REMOVE code Modification.

Added ComponentTracker lines refresh after performing source file modifications.

### CODE REVIEW

File: flow-server/src/main/java/com/vaadin/flow/component/internal/ComponentTracker.java
Comment (sissbruecker) [inline flow-server/src/main/java/com/vaadin/flow/component/internal/ComponentTracker.java:1]: I could still reproduce an error with the following code:
```java
        name = new TextField("Your name");
        name.setValue("World");

        Button sayHello = new Button("Say hello");
        sayHello.addClickListener(e -> Notification.show("Hello " + name.getValue())
        );
        sayHello.addClickShortcut(Key.ENTER);

        setMargin(true);
        setVerticalComponentAlignment(Alignment.END, name, sayHello);

        add(name, sayHello);

        Button another = new Button("Another button");
        add(another);
```

Steps:
- Pick TextField, modify
- Pick "Say hello" button, modify
- Pick "Another button", modify
- Pick "Say hello" button again

That results in:
```
Caused by: java.lang.IllegalStateException: e -> Notification.show("Hello " + name.getValue()) is not StringLiteralExpr, it is LambdaExpr
	at com.github.javaparser.ast.expr.Expression.asStringLiteralExpr(Expression.java:382) ~[javaparser-core-3.25.1.jar:na]
	at com.vaadin.base.devserver.themeeditor.JavaSourceModifier.lambda$getClassNames$1(JavaSourceModifier.java:327) ~[vaadin-dev-server-24.1-SNAPSHOT.jar:na]
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511) ~[na:na]
	at com.github.javaparser.ast.NodeList.forEach(NodeList.java:288) ~[javaparser-core-3.25.1.jar:na]
	at com.vaadin.base.devserver.themeeditor.JavaSourceModifier.lambda$getClassNames$2(JavaSourceModifier.java:326) ~[vaadin-dev-server-24.1-SNAPSHOT.jar:na]
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511) ~[na:na]
	at com.vaadin.base.devserver.themeeditor.JavaSourceModifier.getClassNames(JavaSourceModifier.java:325) ~[vaadin-dev-server-24.1-SNAPSHOT.jar:na]
	at com.vaadin.base.devserver.themeeditor.JavaSourceModifier.lambda$getUniqueClassName$fd7a4367$1(JavaSourceModifier.java:134) ~[vaadin-dev-server-24.1-SNAPSHOT.jar:na]
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539) ~[na:na]
	at java.base/java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:264) ~[na:na]
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java) ~[na:na]
	at com.vaadin.flow.server.VaadinService.runPendingAccessTasks(VaadinService.java:2029) ~[flow-server-24.1-SNAPSHOT.jar:24.1-SNAPSHOT]
	at com.vaadin.flow.server.VaadinSession.unlock(VaadinSession.java:711) ~[flow-server-24.1-SNAPSHOT.jar:24.1-SNAPSHOT]
	at com.vaadin.flow.server.VaadinService.ensureAccessQueuePurged(VaadinService.java:1995) ~[flow-server-24.1-SNAPSHOT.jar:24.1-SNAPSHOT]
	at com.vaadin.flow.server.VaadinService.accessSession(VaadinService.java:1962) ~[flow-server-24.1-SNAPSHOT.jar:24.1-SNAPSHOT]
	at com.vaadin.flow.server.VaadinSession.access(VaadinSession.java:1015) ~[flow-server-24.1-SNAPSHOT.jar:24.1-SNAPSHOT]
	at com.vaadin.base.devserver.themeeditor.JavaSourceModifier.getUniqueClassName(JavaSourceModifier.java:123) ~[vaadin-dev-server-24.1-SNAPSHOT.jar:na]
	... 37 common frames omitted
```

Comment (MarcinVaadin) [reply flow-server/src/main/java/com/vaadin/flow/component/internal/ComponentTracker.java:1]: Fixed, thanks for finding

**Review by sissbruecker — APPROVED** [review]
Could not find issues from manual testing.

Integration with editor logic could use some polishing, but not a blocker.

File: vaadin-dev-server/src/main/java/com/vaadin/base/devserver/themeeditor/JavaSourceModifier.java [hunk 1]
```diff
@@ -67,33 +66,38 @@ public void setClassNames(Integer uiId, Integer nodeId,
...
+                int sourceOffset = 0;
+                List<String> existingClassNames = getClassNames(sourceFile,
+                        createLocation.lineNumber());
+                for (String className : classNames) {
+                    if (!existingClassNames.contains(className)) {
+                        sourceOffset += addComponentAttribute(sourceFile,
+                                createLocation.lineNumber(),
+                                attachLocation.lineNumber(),
+                                ComponentType.BUTTON, "addClassName",
+                                className);
```
[the remark this answers is not in the document]
Comment (MarcinVaadin) [reply vaadin-dev-server/src/main/java/com/vaadin/base/devserver/themeeditor/JavaSourceModifier.java:86]: Well... yes, it does not look good. Some fragments have been already updated and waiting for this PR to be merged. Methods naming is kind of generic as Editor itself is supposed to be used for whole Java classes design.

Scope of that PR was to check if existing Editor can be used, and it has been succeeded, polishing will be done in next PR.

**Pull request merged**